### PR TITLE
Faster source map tracing

### PR DESCRIPTION
**Pull request by jridgewell**
Reimplmements our source map generation to avoid serializing and deserializing the map's JSON (and VLQ encoded mappings) in order to perform tracing. It also makes it much easier to get access to the source map, we aren't dependent on the exact file path details of our source map (they're fingerprinted and high entropy)!

With this, tracing source maps during a client error goes from ~700ms to ~200ms. Still not perfect, but there are other overheads that need to be solved first.

- - -

This PR is based on https://github.com/vercel/turbo/pull/2402. In order to see the code in isolation, open https://github.com/vercel/turbo/compare/pull/2402/head...jridgewell:faster-source-map-tracing

### CODE REVIEW

File: crates/turbopack-core/src/source_map.rs [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+        Ok(BytesVc::cell(bytes))
+    }
+
+    /// Traces a generated line/column into an mapping token representing either
+    /// synthetic code or user-authored original code.
+    #[turbo_tasks::function]
+    pub async fn lookup_token(self, line: usize, column: usize) -> Result<OptionTokenVc> {
+        let token = match &*self.await? {
+            SourceMap::Regular(map) => {
+                match map.lookup_token(line as u32, column as u32) {
```
Comment (alexkirsz) [inline crates/turbopack-core/src/source_map.rs:167]: What's the reasoning for storing line and column as `usize`?

File: crates/turbopack-core/src/source_map.rs [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+        Arc::ptr_eq(&self.0, &other.0)
+    }
+}
+
+/// Wraps the CrateMap struct so that it can be cached in a Vc.
+///
+/// CrateMap contains a raw pointer, which isn't Send, which is required to
+/// cache in a Vc. So, we have wrap it in 4 layers of cruft to do it. We don't
+/// actually use the pointer, because we don't perform sourcesContent lookups,
+/// so it's fine.
```
Comment (alexkirsz) [inline crates/turbopack-core/src/source_map.rs:246]: This disclaimer could go in a `// Safety` comment above the below `unsafe impl`s.

**Review by alexkirsz — APPROVED** [review]

File: crates/turbopack-core/src/source_map.rs (hunk 2 above)
Comment (jridgewell) [reply crates/turbopack-core/src/source_map.rs:246]: Done.

File: crates/turbopack-core/src/source_map.rs (hunk 1 above)
Comment (jridgewell) [reply crates/turbopack-core/src/source_map.rs:167]: `map` here is a `CrateMap`, which expects `u32`. Our internal `SourcePos` is used all over the place, including calling code rust stdlib code, and everything expects `usize` there.

We could probably save bytes by storing everything as a `u32`, but it'll require casting in a lot of existing code.

**Pull request merged**